03-objects-and-composition/exercise/07.js

```diff
@@ -0,0 +1,25 @@
+function townsToJSON(array){
+  const [, ...items] = array
+  //  regexp to match all spaces and |
+  let regex = /\s?\|\s?/;
+  let res = [];
+  
+  for (let i = 0; i <items.length; i++) {
+    let [, town, latitude, longitude,] = items[i].split(regex);
+    latitude = Number(latitude);
+    longitude = Number(longitude);
+    let obj = {};
```

Comment:
Or you can keep the way of formatting but extract it into separate function so to reuse the code:

```javascript
for (let i = 0; i < items.length; i++) {
    const [, town, latitude, longitude] = items[i].split(regex);
    const obj = {
      Town: town,
      Latitude: roundNumber(latitude),
      Longitude: roundNumber(longitude),
    };
    res.push(obj);
  }

  function roundNumber(num) {
    return Math.round(Number(num) * 100) / 100;
  }
```

Reply:
didn't think about this